notebook/services/contents/fileio.py

```diff
@@ -30,6 +30,16 @@
     from base64 import encodestring as encodebytes, decodestring as decodebytes
 
 
+def replace_file(src, dst):
+    """ replace dst with src
+
+    switches between os.replace or os.rename based on python 2.7 or python 3
+    """
+    try:
+        os.replace(src, dst)
+    except:
```

Comment:
In general I think it's best to avoid catching `NameError` and `AttributeError`, because they can both hide typos and other easy mistakes. It's probably not a big deal here, because there's only one attribute in use and it's calling a standard library function (which we can probably assume doesn't have trivial bugs). But it's best to cultivate good habits. ;-)